Dockerfile

```diff
@@ -19,16 +25,42 @@ COPY cmd/ cmd/
 COPY magefiles/ magefiles/
 COPY pkg/ pkg/
 COPY tests/ tests/
-COPY Makefile .
 
-RUN make build
+RUN go install -mod=mod github.com/onsi/ginkgo/v2/ginkgo
+RUN ginkgo build ./cmd
+
+RUN curl -L "https://mirror.openshift.com/pub/openshift-v4/clients/ocp/${OC_VERSION}/openshift-client-linux.tar.gz" -o /tmp/openshift-client-linux.tar.gz && \
+    tar --no-same-owner -xzf /tmp/openshift-client-linux.tar.gz && \
+    mv oc kubectl /usr/local/bin && \
+    oc version --client && \
+    kubectl version --client
+
+RUN curl -L "https://github.com/stedolan/jq/releases/download/jq-${JQ_VERSION}/jq-linux64" -o /usr/local/bin/jq  && \
+    chmod +x /usr/local/bin/jq && \
+    jq --version
+
+RUN curl -L "https://github.com/mikefarah/yq/releases/download/v${YQ_VERSION}/yq_linux_amd64" -o /usr/local/bin/yq && \
+    chmod +x /usr/local/bin/yq && \
+    yq --version
+
+RUN curl -LO "https://github.com/oras-project/oras/releases/download/v${ORAS_VERSION}/oras_${ORAS_VERSION}_linux_amd64.tar.gz" && \
+    mkdir -p oras-install/ && \
+    tar -zxf oras_${ORAS_VERSION}_*.tar.gz -C oras-install/ && \
+    mv oras-install/oras /usr/local/bin/ && \
+    rm -rf oras_${ORAS_VERSION}_*.tar.gz oras-install/ && \
+    oras version
 
-FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
+FROM registry.access.redhat.com/ubi9/go-toolset:1.21.9-1.1717085562
 
+WORKDIR /konflux-e2e
 
+ENV GOBIN=$GOPATH/bin
+ENV E2E_BIN_PATH=/konflux-e2e/konflux-e2e.test
 
-WORKDIR /root/
-COPY --from=builder /github.com/konflux-ci/e2e-tests/bin/e2e-appstudio ./
-COPY --from=builder /github.com/konflux-ci/e2e-tests/tests ./tests
+COPY --from=builder /usr/local/bin/jq /usr/local/bin/jq
+COPY --from=builder /usr/local/bin/yq /usr/local/bin/yq
+COPY --from=builder /usr/local/bin/oc /usr/local/bin/oc
+COPY --from=builder /usr/local/bin/kubectl /usr/local/bin/kubectl
 COPY --from=builder /usr/local/bin/oras /usr/local/bin/oras
-ENTRYPOINT ["/root/e2e-appstudio"]
+COPY --from=builder $GOBIN/ginkgo /usr/local/bin
+COPY --from=builder /konflux-e2e/cmd/cmd.test konflux-e2e.test
```

Comment:
do we have a jira for updating [build-definitions e2e task](https://github.com/konflux-ci/build-definitions/blob/db5c10dce8e72e0644b785fc0defe3e9e531481b/.tekton/tasks/e2e-test.yaml#L28-L37) once this change is in?